docs/proposals/hosted_components_templates.md

```diff
@@ -0,0 +1,559 @@
+installed and what is configured (is what we configured correct for the image
+version we installed).
+
+## Rationale
+As was piloted with the Template Service Broker ("TSB" for short), a role was
+created to process and configure OCP templates that were synchronized (currently
+still a manual process) from the origin repo where the component resides.
+
+This allowed any changes for the installation to be maintained by the developers
+and keep them from needing to be fluent with Ansible. It also afforded a level
+of abstraction from needing to keep in sync with TSB related changes and then
+translate them to changes within an Ansible role.
+
+However, there is still the gap of ensuring a role configures changes appropriately
+for the version of the component it is installing and we are violating the "DRY"
+principle by needing to sync files between repos.
+
+Prior to 3.4, Aggregated Logging solved this using a deployer pod. Everything
+was versioned and contained within this artifact that the Integration
+Services team provided and maintained. The deployer pod lived in the same
+repo as the components that it configured. This is something that is addressed
+again now with the Service Broker, Ansible Playbook Bundles ("APBs" for short)
+are configured in such a way that a containerized install/uninstall is done for
+the component.
+
+## Design
+We should describe a way to create a containerized playbook and role to mirror
+what is done with APBs for hosted components so that the openshift-ansible roles
+would reuse these containers to install/uninstall these components.
+
+The containerized role could follow the pattern used by the TSB within its
+openshift-ansible role `template_service_broker` to maintain a lower learning
+curve for those not familiar with ansible as well as decrease duplication of work
+in cases where the component provides OCP templates already.
+
+Within the openshift-ansible role, installing in a way similar to APBs would
+create an interface that would not need to change once the role is created,
+with the exception being if new information need be passed into the container.
+
+If hosted components adhere to the structure required for the Service Broker
+to provision/deprovision bind/unbind it allows us to also open up the possibility
+to set up these components through the UI. This could further lighten the load
+placed on openshift-ansible and provide an alternative means to install hosted
+components after an OCP installation is completed.
+
+## User Story
+As a developer on OpenShift-Ansible,
+I want roles for hosted components to run the APB container that is provided
+  by the developer of the component
+so that the majority of on-boarding and supporting a hosted component be the
+  responsibility of the developer of the component, reducing the amount of
+  copy and paste, translating to Ansible, and changes within the openshift-ansible
+  role that is done as updates are made.
```

Comment:
I really like this approach in general. With Online team we fought this on a small scale where we wanted ansible to live with the app repos but needed to assemble it for installation. We wrote a small tool to essentially vendor as we would with go, but the container approach sounds better to me.

Reply:
Looking ahead for cluster operator, in the future we hope to have controllers in the operator cluster responsible for maintaining add-on applications like router/registry. We initially imagined that would be via calling different playbooks in openshift-ansible, but calling APB containers would be interesting as well.